### ISSUE TITLE
Question: Advanced - Following module imports

### ISSUE THREAD

**Issue by LukasMachetanz** [madou/typescript-transformer-handbook#8]
First of all; thanks for the handbook; I really appreciate it. Great work.

In the section `Advanced - Following module imports` you write the following:

> You can also traverse the imported node as well using ts.visitChild and the like.

I really would like to know how this could be achieved. I do not understand how I get the imported node and how I can traverse it then.

As a matter of fact I would like to achieve the following:

- I would like to finde a function call like e.g. `transform(PlaygroundComponent)`
- And then I would like to find the corresponding class `PlaygroundComponent` (maybe imported from another file and adjust it)

I would appreciate any help. Thanks. :)

**Comment by madou** [comment]
hi thanks 😄 

i'm not sure if this is possible (in an easy way) to be honest. if they're in the same file you definitely can - you'd just have to traverse the AST twice.

if `PlaygroundComponent` is in a different file, it gets tricky. you would need to do two runs over the code and then figure out how to share data between the two runs. another problem is that transformers get cached in interesting ways (when used with bundlers/test runners) so your files might not be transformed until they are modified.

tldr i've really only traversed imported nodes and then using _that_ information do a some kind of transformation in the current source file in.

hope this helps

**Comment by LukasMachetanz** [comment]
Thank you for the answer. I will take this tipps into consideration. Thank you.

**Comment by LukasMachetanz** [comment]
Just out of curiosity: Isn't the example https://github.com/madou/typescript-transformer-handbook/blob/master/example-transformers/follow-node-modules-imports/transformer.ts going into this direction? Or where is the difference? Couldn't this be a possible approach?

**Comment by madou** [comment]
hello again - that allows us to traverse the import but we can't actually change it

**Comment by LukasMachetanz** [comment]
I see. Thanks anyway. :)
